v2/tools/generator/internal/codegen/pipeline/add_cross_resource_references.go

```diff
@@ -83,20 +83,20 @@ func AddCrossResourceReferences(configuration *config.Configuration, idFactory a
 				// TODO: Properties collapsing work for this.
 			}
 
-			err := kerrors.NewAggregate(crossResourceReferenceErrs)
-			if err != nil {
-				return nil, err
+			if len(crossResourceReferenceErrs) > 0 {
```

Comment:
It looks like `kerrors.NewAggregate` returns nil if it's passed an empty slice - do you need the guard?

Reply:
I wrote it because `err` was being declared as a `kerrors.Aggregate` instead of an `error`; have reworked.